### PR TITLE
Feature/toolboxes refactoring

### PR DESCRIPTION
**Pull request by vincentchabannes**
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/feelpp/feelpp/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes?
- [ ] Have you successfully run the Feel++ testsuite with your changes locally?
- [ ] Have you written Doxygen comments in your contribution ?

-----

* feeldiscr : up operator interpolation by adding a matrix setup struct which provides to give some assembly specification 
* feelmodel and toolboxes: up postprocess measure quantities
* fluid toolbox : refactoring of body articulations and implement the p-matrix method

### CODE REVIEW

**Comment by vincentchabannes** [comment]
@prudhomm can you do the review?

File: toolboxes/feel/feelmodels/levelset/levelset.hpp [hunk 1]
```diff
@@ -409,6 +410,10 @@ class LevelSet : public LevelSetBase<ConvexType, BasisType, PeriodicityType, Bas
         return hana::concat( super_type::allMeasuresQuantities( prefix ), hana::make_tuple(
                     ModelMeasuresQuantity( prefix, "velocity-com", std::bind( eigenToVec, std::bind( &self_type::velocityCOM, this ) ) )
                     ) );
+#endif
+        return Feel::FeelModels::modelMeasuresQuantities( super_type::allMeasuresQuantities( prefix ),
+                                                          modelMeasuresQuantity(prefix, "velocity-com", std::bind( &self_type::velocityCOM, this ) )
```
Comment (prudhomm) [inline toolboxes/feel/feelmodels/levelset/levelset.hpp:415]: could it be more explicit than "velocity-com" ?

Comment (prudhomm) [reply toolboxes/feel/feelmodels/levelset/levelset.hpp:415]: com == center of mass ?

File: toolboxes/feel/feelmodels/modelcore/modelmeasuresquantities.hpp [hunk 2]
```diff
@@ -68,6 +70,86 @@ class ModelMeasuresQuantity
...
+ModelMeasuresQuantity<QuantityType>
+modelMeasuresQuantity( std::string const& prefix, std::string const& name, QuantityType const& q )
+{
+    return ModelMeasuresQuantity<QuantityType>( prefix, name, q );
+}
+
+
+struct ModelMeasuresQuantitiesFeelppTag {};
+
+template <typename TupleQuantityType>
```
Comment (prudhomm) [inline toolboxes/feel/feelmodels/modelcore/modelmeasuresquantities.hpp:101]: doxygen documentation ?

File: toolboxes/feel/feelmodels/modelcore/modelmeasuresquantities.hpp [hunk 3]
```diff
@@ -68,6 +70,86 @@ class ModelMeasuresQuantity
...
+};
+
+template<typename... ModelMeasuresQuantitiesType>
+using model_measures_quantities_t = typename ModelMeasuresQuantitiesTraits<ModelMeasuresQuantitiesType...>::type;
+
+using model_measures_quantities_empty_t = ModelMeasuresQuantities<hana::tuple<>>;
+
+template<typename... ModelMeasuresQuantitiesType>
+model_measures_quantities_t<ModelMeasuresQuantitiesType...>
+modelMeasuresQuantities( const ModelMeasuresQuantitiesType&... smf )
```
Comment (prudhomm) [inline toolboxes/feel/feelmodels/modelcore/modelmeasuresquantities.hpp:147]: doxygen documentation. ?
it seems part of the API

**Review by prudhomm — APPROVED** [review]

File: toolboxes/feel/feelmodels/levelset/levelset.hpp (hunk 1 above)
Comment (vincentchabannes) [reply toolboxes/feel/feelmodels/levelset/levelset.hpp:415]: I don't know, it's level set code of @metivett

Comment (prudhomm) [reply toolboxes/feel/feelmodels/levelset/levelset.hpp:415]: it is probably center of mass.
if we choose this kind of naming, maybe we could use it everywhere and document it

Comment (metivett) [reply toolboxes/feel/feelmodels/levelset/levelset.hpp:415]: Yes it is refers to the "center-of-mass" velocity (or position). Would you prefer another naming ?

**Pull request merged**